client/services.go

```diff
@@ -598,3 +599,9 @@ type WorkspacesClient interface {
 	DescribeWorkspaces(ctx context.Context, params *workspaces.DescribeWorkspacesInput, optFns ...func(*workspaces.Options)) (*workspaces.DescribeWorkspacesOutput, error)
 	DescribeWorkspaceDirectories(ctx context.Context, params *workspaces.DescribeWorkspaceDirectoriesInput, optFns ...func(*workspaces.Options)) (*workspaces.DescribeWorkspaceDirectoriesOutput, error)
 }
+
+//go:generate mockgen -package=mocks -destination=./mocks/mock_codepipeline.go . CodePipelineClient
+type CodePipelineClient interface {
```

Comment:
missing https://docs.aws.amazon.com/codepipeline/latest/APIReference/API_ListTagsForResource.html to get tags for all pipelines